iidm/iidm-api/src/main/java/com/powsybl/iidm/network/LoadingLimitsAdder.java

```diff
@@ -102,6 +102,13 @@ interface TemporaryLimitAdder<A> {
      */
     String getOwnerId();
 
+    /**
+     * <p>Creates a limit that is the exactly same as the one given in parameters.</p>
+     * @param limits the loading limit getting copied.
+     * @return a new limit
+     */
+    A copyFromCurrentLimits(CurrentLimits limits);
```

Comment:
It would be better to simply call

```java
CurrentLimitsAdder adder2 = line.newCurrentLimits2(limits1);
```
instead of 

```java
CurrentLimitsAdder adder2 = line.newCurrentLimits2()
                .copyFromCurrentLimits(limits1);
```
since it doesn't really make sense to call another method before `copyFromCurrentLimits(...)`.

To achieve this, "shortcut" methods should be declared each time an `add...Limits...()` method is defined.

In order to avoid breaking changes, I think the following strategy could be applied:

1. Move the `copyFromCurrentLimits` implementation in `LoadingLimitsUtil`. It should be possible since it relies only on API method calls (note that it should be renamed and should use generic types to ensure that the return's type is compatible with the argument's one — see the previous comments).
2. In all the interfaces where an `add...Limits...()` method is defined, define a **default** method `add...Limits...(...Limits)` which will call the method without argument then call the method defined in 1. on its result.

This way, custom IIDM implementation maintainers won't have to do anything to support this new feature, but it will be possible to them to define a custom implementation if really needed (by overriding the default methods).

Reply:
With the implementations in `AbstractConnectableBranch` and `TieLineImpl`, there is a breaking change for custom IIDM implementation maintainers since they don't use these classes but only the interface (`Branch`).

If you remove these implementations and define the default methods in `Branch`, at is was done for `FlowsLimitsHolder`, there will be no breaking change.